parsing/parser.mly

```diff
@@ -2877,6 +2878,8 @@ pattern:
       { $1 }
   | EXCEPTION ext_attributes pattern %prec prec_constr_appl
       { mkpat_attrs ~loc:$sloc (Ppat_exception $3) $2}
+  | EFFECT pattern_gen COMMA simple_pattern
```

Comment:
We we really want `simple_pattern` ? I would expect only `ident` or `underscore` to be valid. It's probably better to fail during type checking, do we have tests for invalid pattern in that position ?
Also, the manual needs to be updated.

Reply:
No need for a syntactic restriction: this pattern must have type `('a, 'b) continuation`, and `continuation` is an abstract type (no constructor), so only variables and underscore will work.  It's better to let the typechecker handle this, as it produces better error message than the parser.